src/Builder/Select/QueryExpressionVisitor.php

```diff
@@ -0,0 +1,199 @@
+            case Comparison::LT:
+                return sprintf('%s:[* TO %s}', $field, $this->valueToString($value));
+            case Comparison::LTE:
+                return sprintf('%s:[* TO %s]', $field, $this->valueToString($value));
+            case Comparison::RANGE:
+                if (\is_array($value)) {
+                    if (2 === \count($value)) {
+                        return sprintf('%s:[%s TO %s]', $field, $this->valueToString($value[0]), $this->valueToString($value[1]));
+                    }
+
+                    if (1 === \count($value)) {
+                        return sprintf('%s:[%s TO *]', $field, $this->valueToString($value[0]));
+                    }
+                }
+
+                throw new RuntimeException(sprintf('Invalid range value: %s', $value));
+            case Comparison::IN:
+                if (\is_array($value)) {
+                    return sprintf('%s:(%s)', $field, $this->valueToString($value, ' OR ', '"'));
+                }
+
+                return sprintf('%s:%s', $field, $this->valueToString($value, ',', '"'));
+            case Comparison::LIKE:
+            case Comparison::MATCH:
+                if (\is_array($value)) {
+                    return sprintf('%s:(%s)', $field, $this->valueToString($value, ' OR ', '', false));
+                }
+
+                return sprintf('%s:%s', $field, $this->valueToString($value, ',', '', false));
+            case Comparison::NIN:
+                if (\is_array($value)) {
+                    return sprintf('-%s:(%s)', $field, $this->valueToString($value, ' OR ', '"'));
+                }
+
+                return sprintf('-%s:%s', $field, $this->valueToString($value, ',', '"'));
+            case Comparison::REGEXP:
+                if ('/' !== $value[0]) {
+                    $value = sprintf('/%s/', $value);
+                }
+
+                return sprintf('%s:%s', $field, $this->valueToString($value, ',', '', false));
+            case Comparison::EMPTY:
+                return sprintf('(*:* NOT %s:*)', $field);
+            default:
+                throw new RuntimeException(sprintf('Unknown comparison operator: %s', $expression->getOperator()));
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function walkValue(Value $value)
+    {
+        return $value->getValue();
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \Solarium\Exception\RuntimeException
+     */
+    public function walkCompositeExpression(ExpressionInterface $expr)
+    {
+        $comparisons = [];
+
+        foreach ($expr->getComparisons() as $child) {
+            $comparisons[] = $this->dispatch($child);
+        }
+
+        switch ($expr->getType()) {
+            case CompositeComparison::TYPE_AND:
+                return implode(' AND ', $comparisons);
+            case CompositeComparison::TYPE_OR:
+                return implode(' OR ', $comparisons);
```

Comment:
Should we use the constants `\Solarium\QueryType\Select\Query\Query::QUERY_OPERATOR_AND` and `QUERY_OPERATOR_OR` instead? It seems a bit overkillish and I don't think it has any benefits. Just wondering if you had considered it and what made you decide against it.